lib/internal/source_map/source_map_cache.js

```diff
@@ -168,6 +168,11 @@ function sourceMapFromDataUrl(basePath, url) {
 function sourcesToAbsolute(base, data) {
   data.sources = data.sources.map((source) => {
     source = (data.sourceRoot || '') + source;
+    // If the source has a webpack:// scheme, do not attempt to resolve
+    // the path, See: https://github.com/webpack/webpack/issues/9601
+    if (source.includes('webpack://')) {
```

Comment:
If this need special handling I guess something is wrong with the default logic.

An URL with an `webpack://` is per definition an absolute URL. Even if you try to resolve is towards any base (with the URL resolving logic) it will stay identical. Every source URL in the source map should be resolved using `new URL(source, sourceMapLocationAsFileUrl)`, and this should automatically keep webpack urls as they are. No special handling is needed.

Node.js might turn `file://` urls into paths for readability reasons.

Reply:
I do not think we should start including explicit special cases in node core for third party libraries like this.

Reply:
@sokra I understand where your point. I was finding it a bit confusing that your absolute URLs (_`"webpack:///./webpack.js`_) contain what looks like a relative file path (_`./webpack.js`_).

I agree with your interpretation of the spec that it indicates `absolute URLs` should not be resolved 👍 

sorry for the confusion.